src/dbwrapper.cpp

```diff
@@ -170,14 +183,7 @@ CDBWrapper::~CDBWrapper()
 {
     delete pdb;
```

Comment:
In commit "dbwrapper: wrap options in unique_ptr and use DeleteOptions" (6770f066b69c6c271e7096cc647d5d42762f6b05)

I liked the suggestion to make `env` a unique_ptr https://github.com/bitcoin/bitcoin/pull/22663#discussion_r686783500, and maybe `pdb` could be a unique_ptr too?

Reply:
I was not sure how to make `pdb` a unique_ptr and also pass it in to `Open` as `leveldb::DB**` when it is `nullptr` at this point. Maybe I am missing something.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/22663#discussion_r772488258

> I was not sure how to make `pdb` a unique_ptr and also pass it in to `Open` as `leveldb::DB**` when it is `nullptr` at this point. Maybe I am missing something.

No need to do this, but you would need to pass a temporary pointer to Open, and switch to unique_ptr after it returns:

```
leveldb::DB* new_db = nullptr;
leveldb::Status status = leveldb::DB::Open(options, fs::PathToString(path), &new_db);
pdb.reset(new_db);
```